### PR TITLE
Add support to systray & fix icon in BrowserWindows

### PR DESCRIPTION
**Pull request by Mutilador**
I'm using AppImage 0.6.5 with Ubuntu 18.04 and the problem is don't have icon in wallet's window. I fixed

Eu estou usando AppImage 0.6.5 com Ubuntu 18.04 e o problema é que a versão de vocês não aparece o ICON na janela, eu corrigi isso.

### CODE REVIEW

**Comment by pedroferreira1** [comment]
Hey, thanks for the pull request.

I've tried in Windows, Ubuntu (18.04) and macOS and only on macOS the tray icon is really big (see image). I've tried a solution but didn't work, I still need to investigate more but will probably do this later. 
Screenshot 2019-04-15 at 19 40 32

Besides that, I think we should give the option for the user to turn on/off the tray (the default should be off). Maybe you could add a checkbox option in the menu to change it.

**Comment by Mutilador** [comment]
Hi Pedro,

I fixed and add submenu with enable/disable Sistray.

File: public/electron.js [hunk 1]
```diff
@@ -78,6 +101,14 @@ function createWindow () {
     // when you should delete the corresponding element.
     mainWindow = null
   })
+
+  mainWindow.on("close", (e) => {
```
Comment (msbrogli) [inline public/electron.js]: Maybe we should show a message to the user saying that the application is still running on the background (with a checkbox to stop showing this message).

File: public/electron.js [hunk 2]
```diff
@@ -94,6 +125,32 @@ if (process.platform === 'darwin') {
 // Some APIs can only be used after this event occurs.
 app.on('ready', createWindow)
 
+function startSystray() {
+  tray = new Tray(path.join(__dirname, trayIcon));
+  const contextMenu = Menu.buildFromTemplate([
+    {
+      label: 'Show/Hide',
```
Comment (msbrogli) [inline public/electron.js]: Why don't we adjust the label according to the current state? It would be either `Show` or `Hide`.

Comment (msbrogli) [reply public/electron.js]: I prefer `Hide Wallet` and `Show Wallet`.

File: public/electron.js [hunk 3]
```diff
@@ -94,6 +125,32 @@ if (process.platform === 'darwin') {
...
+      click() {  
+        if (mainWindow.isVisible()) {
+          mainWindow.hide();
+        } else {
+          mainWindow.show();
+        }
+      }
+    },
+    { 
+      label: 'Exit',
```
Comment (msbrogli) [inline public/electron.js]: It seems weird to only show the Exit menu when the systray is enabled. Wouldn't it be better to always show it?

Comment (Mutilador) [reply public/electron.js]: > It seems weird to only show the Exit menu when the systray is enabled. Wouldn't it be better to always show it?

This menu is shown when the user clicks on the systray.

File: public/electron.js (hunk 2 above)
Comment (Mutilador) [reply public/electron.js]: Okay, you're right.

File: public/electron.js (hunk 1 above)
Comment (Mutilador) [reply public/electron.js]: done!

File: public/electron.js [hunk 4]
```diff
@@ -17,6 +17,18 @@ Sentry.init({
   release: process.env.npm_package_version
 })
 
+let tray = null
+let iconOS = null;
+let trayIcon = null;
+let msgCheck = false;
```
Comment (msbrogli) [inline public/electron.js]: We should save this variable in local storage. So, we would load it after user restarts the wallet.

File: public/electron.js [hunk 5]
```diff
@@ -94,6 +142,32 @@ if (process.platform === 'darwin') {
 // Some APIs can only be used after this event occurs.
 app.on('ready', createWindow)
 
+function startSystray() {
+  tray = new Tray(path.join(__dirname, trayIcon));
+  const contextMenu = Menu.buildFromTemplate([
+    {
+      label: 'Hide wallet/Show wallet',
```
Comment (pedroferreira1) [inline public/electron.js]: We should set this label as 'Hide Wallet' when `mainWindow.isVisible()` is `true` and 'Show Wallet', otherwise

**Comment by msbrogli** [comment]
@pedroferreira1 It seems it is good to merge. Can you just double check running the code, please?

**Comment by pedroferreira1** [comment]
I've tested here and everything is working fine! I will merge it

**Pull request merged**